addons/hw_posbox_homepage/controllers/main.py

```diff
@@ -258,14 +266,18 @@ def connect_to_server(self, token, iotname):
             token = credential[1]
             db_uuid = credential[2]
             enterprise_code = credential[3]
-            helpers.save_conf_server(url, token, db_uuid, enterprise_code)
+            try:
+                helpers.save_conf_server(url, token, db_uuid, enterprise_code)
+            except (subprocess.CalledProcessError, OSError, Exception):
+                return 'Failed to write server configuration files on IoTBox. Please try again.'
```

Comment:
This error will be displayed to the client on the homepage. (Helps troubleshooting)